source/Weapon.cpp

```diff
@@ -300,18 +317,23 @@ void Weapon::drawDecal(const Point3& point, const Vector3& normal, bool hit) {
 	const shared_ptr<VisibleEntity>& newDecal = VisibleEntity::create(format("decal%03d", ++m_lastDecalID), &(*m_scene), decalModel, decalFrame);
 	newDecal->setCastsShadows(false);
 	m_scene->insert(newDecal);
-	if (!hit) m_currentMissDecals.insert(0, newDecal);	// Add the new decal to the front of the Array (if a miss)
+	if (!hit) {
+		m_currentMissDecals.insert(0, newDecal);	// Add the new decal to the front of the Array (if a miss)
+		m_missDecalTimesRemaining.insert(0, m_config->missDecalTimeoutS);
+	}
 	else {
 		m_hitDecal = newDecal;
 		m_hitDecalTimeRemainingS = m_config->hitDecalDurationS;
 	}
 }
 
-void Weapon::clearDecals() {
-	while (m_currentMissDecals.size() > 0) {
-		m_currentMissDecals.pop();
+void Weapon::clearDecals(bool clearHitDecal) {
+	while (m_currentMissDecals.size() > 0) {				// Remove and clear miss decals
+		m_scene->remove(m_currentMissDecals.pop());
```

Comment:
This line is crashing for me on reload (F5). I assume it's because the scene isn't the same one that the `m_currentMissDecals` assumed it was.

Reply:
Up to now I haven't been able to reproduce this crash. Any other details on repro are welcomed.

Reply:
Since I want to include this in the UL experiment, let's go ahead an merge with this bug pending. It really only impacts developers, so I'm sure we'll hit it and fix it in the future.